### PR TITLE
19257 - Implement Continue In Now button

### PR DESCRIPTION
**Pull request by kzdev420**
*Issue #:*
https://github.com/bcgov/entity/issues/19257

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-auth license (Apache 2.0).

### CODE REVIEW

**Comment by kzdev420** [comment]
To implement these - 
```
If FF is enabled: create draft etc..
If FF is disabled: go to COLIN
```

`Check the FF and set the target`
https://github.com/bcgov/sbc-auth/blob/6e74fda6cb91e468e31363f364a5931c2ee8638c/auth-web/src/stores/business.ts#L100-L107

`Go to the target`
https://github.com/bcgov/sbc-auth/blob/6e74fda6cb91e468e31363f364a5931c2ee8638c/auth-web/src/components/auth/manage-business/AffiliationAction.vue#L278-L297

**Comment by kzdev420** [comment]
I think `legal-api/businesses?draft=true` is not ready for `ContinuationIn`

**Comment by kzdev420** [comment]
https://github.com/bcgov/bcrs-shared-components/blob/0d13caae606ac8f49d1ae9ea986718ba3d5b9de9/src/enums/filing-types.ts#L1

Should I add the `continuationIn` in here too?

**Comment by seeker25** [comment]
package json, package-lock updates please

**Review by seeker25 — APPROVED** [review]
Please fix CI, then
Need someone from entities team to approve before squash -> merge please

File: auth-web/src/components/auth/manage-business/AffiliationAction.vue
Comment (JazzarKarim) [inline auth-web/src/components/auth/manage-business/AffiliationAction.vue:157]: Please add a check here for the feature flag. Should be something similar to what amalgamations do (line 138 and line 153).

File: auth-web/src/stores/business.ts
Comment (JazzarKarim) [inline auth-web/src/stores/business.ts:360]: Shouldn't this be `business.businessIdentifier || business.nameRequest.nrNumber`?

**Comment by JazzarKarim** [comment]
Please also add a check for the "open external link" based on the Feature Flag. Should be line 232.

**Comment by JazzarKarim** [comment]
> https://github.com/bcgov/bcrs-shared-components/blob/0d13caae606ac8f49d1ae9ea986718ba3d5b9de9/src/enums/filing-types.ts#L1
> 
> Should I add the `continuationIn` in here too?

It's already there. You can use the latest version of the enums and install!

**Comment by severinbeauvais** [comment]
> https://github.com/bcgov/bcrs-shared-components/blob/0d13caae606ac8f49d1ae9ea986718ba3d5b9de9/src/enums/filing-types.ts#L1
> 
> Should I add the `continuationIn` in here too?

I'm not sure what you're asking here. It's already in this enum, line 13.

File: auth-web/src/stores/business.ts
Comment (severinbeauvais) [inline auth-web/src/stores/business.ts:343]: Since I'm looking at this code, I think lines 331-333 should be in the filing object (line 325+) not in the registration object. I guess the Legal API just ignores the business object currently.

Comment (severinbeauvais) [reply auth-web/src/stores/business.ts:343]: Never mind. That's for the NOB.

Although, the registration filing object probably should have a business object...

File: auth-web/src/stores/business.ts
Comment (severinbeauvais) [reply auth-web/src/stores/business.ts:360]: This looks right to me.

The business identifier is never a NR number, right? Or are you thinking that the business number might be the description? In that case it would go in the nameRequest.legalName field -- but leave that for a separate ticket to worry about. Here, you're just creating the initial draft.

**Comment by severinbeauvais** [comment]
> I think `legal-api/businesses?draft=true` is not ready for `ContinuationIn`

You're probably right. There's always some initial Legal API work needed to accept a new filing type. Do you guys have a ticket to do that already?

**Comment by severinbeauvais** [comment]
I would love to see some screenshots with a supported Continue In type (create draft) and an unsupported type (go to COLIN).

**Comment by JazzarKarim** [comment]
> > I think `legal-api/businesses?draft=true` is not ready for `ContinuationIn`
> 
> You're probably right. There's always some initial Legal API work needed to accept a new filing type. Do you guys have a ticket to do that already?

Yup, it's https://github.com/bcgov/entity/issues/19261.

File: auth-web/src/stores/business.ts
Comment (JazzarKarim) [reply auth-web/src/stores/business.ts:360]: OK, sounds good Sev. We'll leave it to a separate ticket to flesh that out later. Resolving this conversation. Thanks!

**Review by severinbeauvais — APPROVED** [review]
This is my preliminary approval -- I think my concerns have been addressed but please resolve the issues that Karim identified.

Also, are there any existing unit tests that could be updated?

File: auth-web/src/components/auth/manage-business/AffiliationAction.vue
Comment (JazzarKarim) [reply auth-web/src/components/auth/manage-business/AffiliationAction.vue]: Fixed, thanks! Resolving.

File: auth-web/src/components/auth/manage-business/AffiliationAction.vue
Comment (severinbeauvais) [inline auth-web/src/components/auth/manage-business/AffiliationAction.vue:475]: This looks different than the case for amalgamations (or others). Please follow existing code pattern (or explain why this one is different). Thanks.

File: auth-web/src/components/auth/manage-business/AffiliationAction.vue
Comment (JazzarKarim) [reply auth-web/src/components/auth/manage-business/AffiliationAction.vue]: If possible, please let's follow a similar pattern to amalgamations just as Sev mentioned in his latest comment. Maybe a `goToContinuationIn()` function?

File: auth-web/src/stores/business.ts
Comment (severinbeauvais) [reply auth-web/src/stores/business.ts:343]: Kevin, can you fix this along the way, please?

File: auth-web/src/components/auth/manage-business/AffiliationAction.vue
Comment (severinbeauvais) [reply auth-web/src/components/auth/manage-business/AffiliationAction.vue]: Looks great, Kevin! 👍

File: auth-web/src/components/auth/manage-business/AffiliationAction.vue
Comment (JazzarKarim) [reply auth-web/src/components/auth/manage-business/AffiliationAction.vue]: Fixed, thanks! Resolving.

**Review by JazzarKarim — APPROVED** [review]
LGTM 👍

File: auth-web/src/models/business.ts
Comment (severinbeauvais) [inline auth-web/src/models/business.ts:118]: No, NOB doesn't need to be here.

File: auth-web/src/components/auth/manage-business/AffiliationAction.vue
Comment (kzdev420) [reply auth-web/src/components/auth/manage-business/AffiliationAction.vue:285]: Yes, they are different.

File: auth-web/src/stores/business.ts
Comment (severinbeauvais) [inline auth-web/src/stores/business.ts:329]: The draft registration filing should have 2 business objects.

One at the top level, here, containing just legalType.

And the other under the registration object as in the previous code, containing just the NOB.

File: auth-web/src/components/auth/manage-business/AffiliationAction.vue
Comment (severinbeauvais) [reply auth-web/src/components/auth/manage-business/AffiliationAction.vue:285]: Which one is correct -- did the previous code work, or does this code work?

File: auth-web/src/models/business.ts
Comment (severinbeauvais) [reply auth-web/src/models/business.ts]: Fixed, thanks.

File: auth-web/src/components/auth/manage-business/AffiliationAction.vue
Comment (kzdev420) [reply auth-web/src/components/auth/manage-business/AffiliationAction.vue:285]: I thought this part was right for checking the FF.

https://github.com/bcgov/sbc-auth/blob/6e74fda6cb91e468e31363f364a5931c2ee8638c/auth-web/src/stores/business.ts#L100-L107

So, I used the `useNameRequest` to go to target

https://github.com/bcgov/sbc-auth/blob/6e74fda6cb91e468e31363f364a5931c2ee8638c/auth-web/src/components/auth/manage-business/AffiliationAction.vue#L278-L297

File: auth-web/src/components/auth/manage-business/AffiliationAction.vue
Comment (severinbeauvais) [reply auth-web/src/components/auth/manage-business/AffiliationAction.vue]: Fixed, thanks.

File: auth-web/src/components/auth/manage-business/AffiliationAction.vue
Comment (JazzarKarim) [reply auth-web/src/components/auth/manage-business/AffiliationAction.vue:285]: Unless I'm missing something, new code should be OK.

Comment (severinbeauvais) [reply auth-web/src/components/auth/manage-business/AffiliationAction.vue:285]: Your first code example uses `enableBcCccUlc`, which is the feature flag for incorporation applications:

image

I think your new code is probably correct, but please test it well with different combinations for both `enable-bc-ccc-ulc` and `supported-amalgamation-entities`. Watch for going to the dashboard vs going to COLIN accordingly. (You can test locally by overriding the FF values.)

**Review by severinbeauvais — APPROVED** [review]
I approve on the assumption that testing shows that it is working correctly with the different feature flags (per comment above).

**Comment by seeker25** [comment]
Is there a release date for this? I'll have to create a release ticket and tag this one there. I think we're backed up by QA atm for NSF.

File: auth-web/src/components/auth/manage-business/AffiliationAction.vue
Comment (JazzarKarim) [reply auth-web/src/components/auth/manage-business/AffiliationAction.vue:285]: @kzdev420 Kevin were you able to test so that we can merge this and unblock Travis?

Comment (kzdev420) [reply auth-web/src/components/auth/manage-business/AffiliationAction.vue:285]: Yes, I tested. It was good for me.
I am going to merge the PR if you are okay

Comment (JazzarKarim) [reply auth-web/src/components/auth/manage-business/AffiliationAction.vue:285]: Okay with me!

**Pull request merged**
